core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -0,0 +1,346 @@
+package edu.asu.c3simulator.widgets;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.badlogic.gdx.Game;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+/**
+ * This class creates a navigation panel by creating the buttons and linking them with
+ * their screens. It also supports having one level of sub-buttons to a parent button. A
+ * parent button will only show and hide sub-buttons when clicked. Only one of the parent
+ * buttons can show its sub-buttons at the same time. If a parent button is clicked, the
+ * already extended sub-buttons will collapse before extending the new set of sub-buttons.
+ * 
+ * @author Alyahya, Mohammed
+ */
+public class NavigationPanel extends Table
+{
+	private class NavigationButtonListener extends ClickListener
+	{
+		String screenName;
+		Screen targetedScreen;
+		
+		public NavigationButtonListener(String screenName, Screen targetedScreen)
+		{
+			this.screenName = screenName;
+			this.targetedScreen = targetedScreen;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			System.out.println(screenName);
+			game.setScreen(targetedScreen);
+		}
+	}
+	
+	private class CollapsibleButtonListener extends ClickListener
+	{
+		String screenName;
+		
+		public CollapsibleButtonListener(String screenName)
+		{
+			this.screenName = screenName;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			if (displayedCollapsibleButton.equalsIgnoreCase(screenName))
+				displayedCollapsibleButton = "";
+			else
+				displayedCollapsibleButton = screenName;
+			recreateNavigationPanel();
+		}
+	}
+	
+	private Game game;
+	private Skin skin;
+	private LinkedHashMap<String, Screen> buttons;
+	private LinkedHashMap<String, LinkedHashMap<String, Screen>> subButtons;
```

Comment:
1: What do you mean "**easier for the user of this class**?" The hash maps are a private implementation detail, so the user of this class shouldn't know about or have access to them...

2: "**use it as much as possible**" Ah. So you want the user to be able to lookup button-subButton relationships without clicking the button? 

2.1: In that case, a tree might be a better choice then. It would give you access to the raw data, but reduce the members needed to track this from 3 (3 hash maps) to 1 (a single tree)

3: "**the user can add buttons any time and remove them and manipulate them based only on the button's name**" This is possible with button listeners too, just keep a single hash map linking buttonName to buttonListener

3.1: I was under the impression that all NavigationPanels will expand and collapse sub buttons automatically. Does this still stand?

3.2: I was also under the impression that all NavigationPanels will redirect the game to the specified Screen when a linked button is clicked. Does this still stand, or does the user have to add that (and other) functionality?

4: "**the way the code creates the navigation panel right now depends on them**" That is concerning. Will you point me to an example? I'm having a hard time imagining how code outside of this class would depend on private hash maps in this class. 

5: After we get the above points ironed out, I can spend some time refactoring, and finishing up this branch so it can be merged into master_testing, and master soon after. Your time would probably be better spent on getting some more features out.